### PR TITLE
Updated to fix compilation using mingw on Windows.

### PR DESCRIPTION
**Pull request by pkulchenko**
This fixes mingw/gcc compilation issues for me. Related ticket #78.

### CODE REVIEW

File: src/os_dependent/FSNodePOSIX.cxx [hunk 1]
```diff
@@ -230,5 +230,10 @@ bool AbstractFilesystemNode::dirExists(const string& path)
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 bool AbstractFilesystemNode::makeDir(const string& path)
 {
-  return mkdir(path.c_str(), 0777) == 0;
+  return mkdir(path.c_str()
```
Comment (mgbellemare) [inline src/os_dependent/FSNodePOSIX.cxx]: I believe `FSNodePOSIX.cxx` shouldn't be compiled under Windows, and looking at your change to CMakeLists.txt that should also be true for mingw. Can you verify that this edit is still necessary?

**Comment by pkulchenko** [comment]
@mgbellemare, you are correct, the change in `FSNodePOSIX.cxx` is no longer needed. I tested with the latest master and it works as expected. Rebased and pushed.

**Comment by mgbellemare** [comment]
LGTM. Thanks for the fix.

**Comment by Jragonmiris** [comment]
Is there any way this can also be routed into the dev branch somehow? I'm trying to use the dev branch, but this is only on master and it's a bit wonky since I get to basically choose whether it compiles or uses new features.

**Comment by pkulchenko** [comment]
I wasn't even aware there was a dev branch. @mgbellemare, what's the difference between the dev and the master branches?

**Comment by mgbellemare** [comment]
We've been using the dev branch for tagged releases. A number of improvements made their way directly into master, with the unfortunate side-effect of having slightly different features in both. We're hoping to merge this weekend, so everything should be sorted then.

**Pull request merged**